shiny/ui/_chat.py

```diff
@@ -1325,6 +1376,210 @@ async def _send_custom_message(self, handler: str, obj: ClientMessage | None):
             },
         )
 
+    @overload
+    def enable_bookmarking(
+        self,
+        client: ClientWithState,
+        /,
+        *,
+        bookmark_on: Optional[Literal["response"]] = "response",
+    ) -> CancelCallback: ...
+    @overload
+    def enable_bookmarking(
+        self,
+        client: ChatlasClient[Any, Any],
+        /,
+        *,
+        bookmark_on: Optional[Literal["response"]] = "response",
+    ) -> CancelCallback: ...
+    def enable_bookmarking(
+        self,
+        client: Any,
```

Comment:
What's the benefit of this `overload` approach over `client: ChatlasClient[Any, Any] | ClientWithState`?

Reply:
Using your approach